root/static/scripts/common/components/CommonsImage.js

```diff
@@ -7,53 +7,63 @@
  * later version: http://www.gnu.org/licenses/gpl-2.0.txt
  */
 
+import * as Sentry from '@sentry/browser';
 import * as React from 'react';
 
 import {minimalEntity} from '../../../../utility/hydrate.js';
 import entityHref from '../utility/entityHref.js';
 
-type Props = {
-  +cachedImage: ?CommonsImageT,
-  +entity: NonUrlRelatableEntityT,
-};
-
-type State = {
-  image: ?CommonsImageT,
-};
-
-class CommonsImage extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {image: props.cachedImage};
-  }
-
-  componentDidMount() {
-    if (!this.state.image) {
-      const $ = require('jquery');
-      $.get(entityHref(this.props.entity, '/commons-image'), data => {
-        this.setState({image: data.image});
-      });
+type CommonsImageRequestCallbackT = (CommonsImageT | null) => void;
+
+function loadCommonsImage(
+  entity: NonUrlRelatableEntityT,
+  callback: CommonsImageRequestCallbackT,
+): void {
+  const url = entityHref(entity, '/commons-image');
+
+  fetch(url)
+    .then(resp => resp.json())
+    .then((reqData) => {
+      callback(reqData.image);
+    })
+    .catch((error) => {
+      console.error(error);
+      Sentry.captureException(error);
+    });
+}
+
+component CommonsImage(
+  cachedImage: ?CommonsImageT,
+  entity: NonUrlRelatableEntityT,
+) {
+  const [commonsImage, setCommonsImage] = React.useState(cachedImage);
+
+  const loadCallback: CommonsImageRequestCallbackT =
+    React.useCallback((data) => {
+      setCommonsImage(data);
+    }, [setCommonsImage]);
+
+  React.useEffect(() => {
+    if (cachedImage == null) {
+      loadCommonsImage(entity, loadCallback);
```

Comment:
The `set` function returned from `useState` [is stable](https://react.dev/reference/react/useState#setstate), so there's actually no need to wrap it in `useCallback`. You don't even have to include it in `useEffect` dependency lists.
```suggestion
  React.useEffect(() => {
    if (cachedImage == null) {
      loadCommonsImage(entity, setCommonsImage);
```